src/test/java/com/pretz/dyntreeserver/controller/DynTreeValidationTest.java

```diff
@@ -0,0 +1,271 @@
+package com.pretz.dyntreeserver.controller;
+
+import com.pretz.dyntreeserver.service.dto.DynTreeDTO;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import java.time.Year;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DynTreeValidationTest {
+
+    private DynTreeDTO testDynTreeDTO;
+    private static Validator validator;
+
+    @BeforeAll
+    public static void createValidator() {
+        ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+        validator = validatorFactory.getValidator();
+    }
+
+    @Test
+    public void shouldPassValidation() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName("Duda")
+                .familyCount(10)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
+
+        Set<ConstraintViolation<DynTreeDTO>> violations
+                = validator.validate(testDynTreeDTO);
+
+        assertTrue(violations.isEmpty());
+    }
+
+    @Test
+    public void shouldFailValidationBlankFamilyName() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName("    ")
+                .familyCount(10)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
+
+        Set<ConstraintViolation<DynTreeDTO>> violations
+                = validator.validate(testDynTreeDTO);
+
+        assertFalse(violations.isEmpty());
+        assertEquals(1, violations.size());
+        ConstraintViolation violation = violations.iterator().next();
+        assertEquals("must not be blank", violation.getMessage());
+        assertEquals("familyName", violation.getPropertyPath().toString());
+        assertEquals("    ", violation.getInvalidValue());
+    }
+
+    @Test
+    public void shouldFailValidationEmptyFamilyName() {
```

Comment:
isn't this case included in the test above?

your `assertEquals` look 99% the same - and I'd say what was the original invalid value isn't really that important?

Reply:
Because I added checking blank inputs at the very end. Yes, there are very similar... but! I understand that empty value can be sent by some custom UI mumbo-jumbo while blank only when end user fills in this field with spaces because of reasons